Lib/configparser.py

```diff
@@ -692,6 +700,10 @@ def add_section(self, section):
         if section == self.default_section:
             raise ValueError('Invalid section name: %r' % section)
 
+        if section is UNNAMED_SECTION:
+            if not self._allow_unnamed_section:
+                raise UnnamedSectionDisabledError()
```

Comment:
You could omit parentheses when raising an exception without argument, but I'm not sure if this is what we usually do (so maybe check the rest of the module / rest of the library).